crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -1255,6 +1255,13 @@ async fn test_handle_certificate_with_shared_object_interrupted_retry() {
         std::mem::drop(g);
 
         // Now run the tx to completion
+        //
+        // NOTE: this test fails if execute_certificate() is used here instead, while keeping
+        // execute_certificate_internal() above. This is because owned object locks are not cleared
+        // atomically after a transaction completes, causing inconsistency when TransactionManager
+        // calss get_missing_input_objects(). This would be an issue for crash recovery too.
```

Comment:
I don't think i understand this issue - can you explain more? What's the nature of the bug?

Reply:
ah, i see. this is a genuine issue then. The reason that `execute_certificate_internal` can tolerate this inconsistency is that it is designed to: See #6173 for background.

I'm trying to think if we can avoid having TxMgr be aware of partial execution.